### PR TITLE
fix(maa): 修复托管下借战赚信用与访问好友每日去重失效

### PR DESCRIPTION
**Pull request by jinghero**
## 摘要

- MAA 的信用收支原生提供「借战赚信用一天仅一次」「访问好友一天仅一次」的去重选项，但其状态保存在 MAA 目录配置中；MAS 每次代理后会还原该目录，导致这两个原生选项在 MAS 托管场景下失效——同一天多次托管会重复执行，单个用户每次多耗时约 4 分钟（借战约 3 分钟、访问好友约 45 秒），且不产生额外收益（OF-1 的每日信用加成仅首次有效）
- 本修复适配该原生功能：改由 MAS 按用户记录执行日期（按游戏换日 UTC+4），当天已执行过则在任务参数中关闭对应子项，去重语义与 MAA 原生选项保持一致
- 领取信用与信用购物不受影响；多用户/多账号各自独立记录；固定关卡与计划表两种配置模式、脚本/用户两种配置来源下行为一致

## 验证

- 真机验证：同日第一次托管正常执行借战与访问好友并记录；第二次托管日志出现「本次跳过借战赚信用/本次跳过访问好友」，MAA 队列不再进入 OF-1 与好友访问
- 单元测试覆盖换日边界（游戏日按 UTC+4 计算）

## Sourcery 摘要

为 MAS 托管场景持久化信用收支每日任务的完成状态，避免重复借战和访问好友。

Bug 修复：
- 在 MAS 托管模式下恢复借战赚信用和访问好友的每日去重，避免同一游戏日重复执行。

改进：
- 按用户和账号独立保存两项任务的最近完成日期，并以 UTC+4 游戏日判断是否跳过任务。

测试：
- 新增覆盖日期去重及 UTC+4 换日边界的单元测试。

<details>
<summary>Original summary in English</summary>

## Sourcery 摘要

恢复托管运行期间 MAA 信用收益任务可靠的每日去重功能。

错误修复：
- 恢复 MAS 托管的 MAA 运行中信用作战和访问好友任务的每日去重，防止在同一个游戏日内重复执行。

增强功能：
- 为每个用户和账号分别持久化完成日期，并使用 UTC+4 的游戏日边界来判断是否应跳过每日任务。

测试：
- 添加覆盖每日去重和 UTC+4 日期边界行为的单元测试。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Restore reliable daily deduplication for MAA credit income tasks during managed runs.

Bug Fixes:
- Restore daily deduplication for credit-fighting and friend-visit tasks in MAS-managed MAA runs, preventing repeated execution within the same game day.

Enhancements:
- Persist completion dates independently for each user and account, using the UTC+4 game-day boundary to determine whether daily tasks should be skipped.

Tests:
- Add unit tests covering daily deduplication and UTC+4 date-boundary behavior.

</details>

</details>

### CODE REVIEW

File: app/task/MAA/AutoProxy.py [hunk 1]
```diff
@@ -101,6 +101,33 @@ def _current_month_marker(now: datetime) -> str:
...
 
 
+def _is_marked_today(completed_date: str, now: datetime) -> bool:
+    """判断「每天一次」类完成标记是否已落在本游戏日（UTC+4）内。
+
+    MAS 托管会还原 MAA 目录，MAA 自带的每日去重（如借战、访问好友）
+    不跨运行生效，因此由 MAS 在日志中检测完成后自行记录。
+    """
+
+    return completed_date == now.strftime("%Y-%m-%d")
```
[the remark this answers is not in the document]
Comment (jinghero) [reply app/task/MAA/AutoProxy.py]: 已按建议修复：`_is_marked_today` 内部将 `now` 归一到 UTC+4（`now.astimezone(UTC4).strftime("%Y-%m-%d")`），不再依赖调用方传入特定时区的 datetime；补充了 UTC 临界时刻的单元测试。见提交 8c3afec3。
